client/service/src/builder.rs

```diff
@@ -1087,16 +1059,19 @@ ServiceBuilder<
 				"disk_write_per_sec" => info.usage.as_ref().map(|usage| usage.io.bytes_written).unwrap_or(0),
 			);
 			if let Some(metrics) = metrics.as_ref() {
-				metrics.node_memory.set(memory);
-				metrics.node_cpu.set(f64::from(cpu_usage));
-				metrics.tx_count.set(txpool_status.ready as u64);
-				metrics.finality_height.set(finalized_number);
-				metrics.best_height.set(best_number);
-				metrics.peers_num.set(num_peers as u64);
-				metrics.node_download.set(net_status.average_download_per_sec);
-				metrics.node_upload.set(net_status.average_upload_per_sec);
+				metrics.memory_usage_bytes.set(memory);
+				metrics.cpu_usage_percentage.set(f64::from(cpu_usage));
+				metrics.transactions_total.inc_by(txpool_status.ready as u64);
```

Comment:
If I am not mistaken [txpool_status.ready](https://github.com/paritytech/substrate/blob/master/primitives/transaction-pool/src/pool.rs#L42) is not being reset after this call, right? If we have two scrapes and thus call `metrics.transactions_total.inc_by(txpool_status.ready as u64);` twice without a reset, the number will not be correct.

Reply:
In addition the `Number of transactions in the ready queue` is not a monotonically increasing number. Thus a counter would not be correct, but a gauge would need to be used.

Reply:
Also does this really represent the `"Total number of transactions"` or just the number of transactions that are currently in the ready queue?

```rust
	/// Number of transactions in the ready queue.
	pub ready: usize,
```

Reply:
Hmm, yeah I think just a normal gauge would be best then.

Reply:
Done